pkg/objstore/objtesting/foreach.go

```diff
@@ -0,0 +1,68 @@
+package objtesting
+
+import (
+	"os"
+	"testing"
+
+	"time"
+
+	"github.com/fortytw2/leaktest"
+	"github.com/improbable-eng/thanos/pkg/objstore"
+	"github.com/improbable-eng/thanos/pkg/objstore/gcs"
+	"github.com/improbable-eng/thanos/pkg/objstore/inmem"
+	"github.com/improbable-eng/thanos/pkg/objstore/s3"
+	"github.com/improbable-eng/thanos/pkg/testutil"
+)
+
+// ForeachStore runs given test using all available objstore implementations.
+// For each it creates a new bucket with a random name and a cleanup function
+// that deletes it after test was run.
+func ForeachStore(t *testing.T, testFn func(t testing.TB, bkt objstore.Bucket)) {
```

Comment:
Is this also copypasted? If no, I don't think we should be using env variables. We should have instead:
`pkg/compact/e2e_test.go` exporting fn `testE2E(t *testing.T, s Store)`
`pkg/objtesting/acceptance_test.go` exporting fn `testAcceptance(t *testing.T, s Store)`
`foreach.go` exporting `testAll`:
```
func testAll(t *testing.T, s Store) {
  - run e2e test as subtest
  - run acceptance test as subtest
  ...
}
```

and call testAll from s3/ package, gcs package, etc.

Reply:
No, it is new logic, let's talk offline, maybe your idea is better.